lightning_client.go

```diff
@@ -2703,3 +2708,60 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// InvoiceSubscription holds the parameters to specify from which update
+// to start streaming
+type InvoiceSubscription struct {
+	//
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all added indexes with an add_index greater than this
+	//value. This allows callers to catch up on any events they missed while they
+	//weren't connected to the streaming RPC.
+	AddIndex uint64
+	//
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all settled indexes with an settle_index greater than
+	//this value. This allows callers to catch up on any events they missed while
+	//they weren't connected to the streaming RPC.
+	SettleIndex uint64
+}
+
+// SubscribeInvoices allows a client to subscribe to updates
+// of newly added/settled invoices.
+func (s *lightningClient) SubscribeInvoices(ctx context.Context, req InvoiceSubscription) (<-chan *lnrpc.Invoice, <-chan error, error) {
```

Comment:
nit: please wrap comments and code at 80 characters (with using 8 characters per tabulator), also see https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md

Reply:
I was a little bit too worried by the returned channel I forgot to format that line. Sorry I've fixed it.